src/timeline/TimelineModel.cpp

```diff
@@ -272,6 +273,33 @@ TimelineModel::data(const QString &id, int role) const
                 return QVariant(toRoomEventType(event));
         case TypeString:
                 return QVariant(toRoomEventTypeString(event));
+        case EmojiCount: {
+                QString qBody = QString::fromStdString(body(event));
+
+                QVector<uint> utf32_string = qBody.toUcs4();
+                bool insideFontBlock       = false;
+                int emojiCount             = 0;
+
+                for (auto &code : utf32_string) {
+                        // TODO: Be more precise here.
+                        if ((code >= 0x2600 && code <= 0x27bf) ||
```

Comment:
Please write a common codepointIsEmoji function and reuse that here and in isEmoji

Reply:
I assume by isEmoji, you meant replaceEmoji. The code you are referencing is written so that it also checks if the text is ONLY emoji and returns "0" by default, if it isn't.